model/src/main/proto/arguments.proto

```diff
@@ -326,3 +327,516 @@ message SurveyActivityParams {
   // The ID of the triggering exploration.
   string exploration_id = 3;
 }
+
+// Represents the parameters needed to create TopicActivity.
+message TopicActivityParams {
+  // The ID of the topic to be loaded.
+  string topic_id = 1;
+
+  // The ID of the story to to be loaded.
+  string story_id = 2;
+}
+
+// Arguments required when creating a TopicFragment.
+message TopicFragmentArguments {
+  // The ID of the topic to be loaded.
+  string topic_id = 1;
+
+  // The ID of the story to to be loaded.
+  string story_id = 2;
+}
+
+// Represents the parameters needed to create QuestionPlayerActivity.
+message QuestionPlayerActivityParams {
+  // Skill ID list for practice.
+  repeated string skill_ids = 1;
+}
+
+// The bundle of properties that are saved upon configuration changes in AdministratorControlActivity.
+message AdministratorControlActivityStateBundle {
+
+  // Title for the loaded fragment.
+  string selected_controls_title = 1;
+
+  // Represents last loaded fragment.
+  string last_loaded_fragment = 2;
+
+  // Represents profile deletion dialog visibility.
+  bool is_profile_deletion_dialog_visible = 3;
+
+  // Represents the profile ID of the current profile.
+  int32 selected_profile_id = 4;
+}
+
+// Arguments required when creating an AdministratorControlsFragment.
+message AdministratorControlsFragmentArguments {
+  // The value indicates whether the layout is multipane or not.
+  bool is_multipane = 1;
+}
+
+// Represents the parameters needed to create MarkChaptersCompletedTestActivity.
+message MarkChaptersCompletedTestActivityParams{
+  // The ID of the profile that wants to mark chapters as completed.
+  int32 internal_profile_id = 1;
+
+  // The value indicates whether to show a confirmation dialog or not.
+  bool show_confirmation_notice = 2;
+}
+
+// Represents the parameters needed to create MarkChaptersCompletedActivity.
+message MarkChaptersCompletedActivityParams{
+  // The ID of the profile that wants to mark chapters as completed.
+  int32 internal_profile_id = 1;
+
+  // The value indicates whether to show a confirmation dialog or not.
+  bool show_confirmation_notice = 2;
+}
+
+// Arguments required when creating a MarkChaptersCompletedFragment.
+message MarkChaptersCompletedFragmentArguments{
+  // The ID of the profile that wants to mark chapters as completed.
+  int32 internal_profile_id = 1;
```

Comment:
Here & elsewhere in this proto: could we perhaps use ProfileId instead? That's the structure meant for passing around a profile ID rather than exposing internal_profile_id directly.

Reply:
If i directly include ProfileId in proto, there could be many changes in code, like in activity creation methods, we passed internalProfileId(Int type) as an argument,
We have two options 
1. Either i create ProfileId using internalProfileId inside method (and pass it to proto and wrap it in bundle).
2. or i change all the activity or fragment creation method to pass ProfileId instead of internalProfileId(Int type) to avoid option 1.

But here  this issue [#4865](https://github.com/oppia/oppia-android/issues/4865) already filed, which will remove all ProfileId from proto in future and use **CurrentUserProfileIdIntentDecorator**  to pass ProfileId.

Thats why i kept internalProfileId in proto. Can you please tell whether i should keep as it is or replace internalProfileId(int type) from ProfileId type.

Reply:
I think that the current approach is a reasonable stop-gap, so I'm happy to resolve this.